imagekit-sdk/src/main/java/io/imagekit/sdk/tasks/UrlGen.java

```diff
@@ -95,31 +111,24 @@ private static String buildPathUrl(String path, String urlEndpoint, Map<String,
         }
 
         URI baseUrl=URI.create(urlEndpoint);
-        URI newUri= null;
-        try {
+        URL newUri= null;
             String newPath=baseUrl.getPath();
-            if (tr.toString().equalsIgnoreCase("")){
-                newPath+=path.substring(1,path.length());
-            }
-            else {
-                newPath+=tr+path;
+            if (tr.toString().equalsIgnoreCase("")) {
+                newPath += path.substring(1);
+            } else {
+                newPath += tr + path;
             }
 
-            URI tmpUri = new URI(baseUrl.getScheme(),baseUrl.getUserInfo(),baseUrl.getHost(),baseUrl.getPort(),
-                    newPath,
-                    queryMaker.get(),null);
+            URL tmpUri = new URL(baseUrl.getScheme(), baseUrl.getHost(), baseUrl.getPort(),
```

Comment:
variable should be tmpUrl and maybe include a comment about why you are doing it this way

Reply:
Renamed and added comment.
👍